pkg/config/config.go

```diff
@@ -264,11 +269,39 @@ func (c *Config) GetConfiguredSrcAddr() (net.IP, error) {
 		return nil, nil
 	}
 
-	return GetInterfaceAddr(*c.Defaults.SrcInterface)
+	ipVersion := c.GetIPVersion()
+
+	return GetInterfaceAddr(*c.Defaults.SrcInterface, ipVersion)
+}
+
+func (c *Config) GetIPVersion() uint8 {
+	strValue := *c.SrcRange
+	ip ,_ , err := net.ParseCIDR(strValue)
+	if ip == nil {
+		ip = net.IP(*c.Defaults.SrcRange)
+	}
+
+	if err != nil {
+		panic("No source range defined")
```

Comment:
Well, might also be just an illegal "prefix". Should definitely print the error here.
Actually I think this conversion should happen very early so it can't fail here.